gobblin-api/src/main/java/gobblin/configuration/WorkUnitState.java

```diff
@@ -171,6 +172,14 @@ public void setHighWaterMark(long value) {
   }
 
   @Override
+  public Properties getProperties() {
+    Properties props = new Properties();
+    props.putAll(this.workunit.getProperties());
+    props.putAll(super.getProperties());
```

Comment:
this.getProperties is an infinite recursion, and this.properties is private. We can either make this.properties protected, or leave it as it is right now. I have no preference.